### PR TITLE
add doc for tidb_last_query_info and tidb_shard_allocate_step

### PR DESCRIPTION
**Pull request by tangenta**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.2 (TiDB 6.2 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [x] v5.4 (TiDB 5.4 versions)
- [x] v5.3 (TiDB 5.3 versions)
- [x] v5.2 (TiDB 5.2 versions)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tidb/pull/21557, https://github.com/pingcap/tidb/pull/18181

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -1964,6 +1974,14 @@ explain select * from t where age=5;
...
 - Note that if `SHARD_ROW_ID_BITS` and `PRE_SPLIT_REGIONS` have been set when a table is created, the specified number of Regions are evenly split after the table creation.
 
+### `tidb_shard_allocate_step` <span class="version-mark">New in v5.0</span>
+
+- Scope: SESSION | GLOBAL
+- Persists to cluster: Yes
+- Default value: `9223372036854775807`
+- Range: `[1, 9223372036854775807]`
+- This variable controls the max number of the continous IDs. Generally, the allocating [`AUTO_RANDOM`](/auto-random.md) IDs or the [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md) annotated row IDs are incremental and continuous in one transaction. This variable can be used to solve the hotspot issue in the large transaction scenario.
```
Comment (bb7133) [inline system-variables.md:1983]: > max number of the continous IDs

Should it be 'max number of the continuous IDs in one single SQL statement'?

Comment (bb7133) [inline system-variables.md:1983]:
```suggestion
- This variable controls the max number of the continuous IDs. Generally, the allocating [`AUTO_RANDOM`](/auto-random.md) IDs or the [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md) annotated row IDs are incremental and continuous in one SQL statement. This variable can be used to solve the hotspot issue in the large transaction scenario.
```
Ditto.

File: system-variables.md [hunk 2]
```diff
@@ -1470,6 +1470,16 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
...
 - This variable is used to set the storage engine list that TiDB can use when reading data.
 
+### `tidb_last_query_info` <span class="version-mark">New in v4.0.14</span>
+
+- Scope: SESSION
+- Default value: ""
+- This is a read-only variable. It is used to query the transaction information of the last DML statement. The information include:
+    - `txn_scope`: The scope of the transaction. It can be `global` or `local`.
+    - `start_ts`: The start timestamp of the transaction.
+    - `for_update_ts`: The for_udpate timestamp of the transaction.
```
Comment (bb7133) [inline system-variables.md:1480]:
```suggestion
    - `for_update_ts`: The for_update_ts of the previously executed DML statement, which is an internal term of TiDB used for tests, usually, the end-user doesn't need to care about it.
```

File: system-variables.md [hunk 3]
```diff
@@ -1470,6 +1470,16 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
 - Default value: `tikv,tiflash,tidb`
 - This variable is used to set the storage engine list that TiDB can use when reading data.
 
+### `tidb_last_query_info` <span class="version-mark">New in v4.0.14</span>
+
+- Scope: SESSION
+- Default value: ""
+- This is a read-only variable. It is used to query the transaction information of the last DML statement. The information include:
```
Comment (bb7133) [inline system-variables.md:1477]:
```suggestion
- This is a read-only variable. It is used to query the transaction information of the last DML statement. The information includes:
```

File: system-variables.md (hunk 1 above)
Comment (tangenta) [reply system-variables.md:1983]: Nope. It should be one transaction:

```sql
mysql> create table t (a bigint auto_random primary key, b int);
Query OK, 0 rows affected, 1 warning (0.04 sec)

mysql> set @@tidb_shard_allocate_step = 3;
Query OK, 0 rows affected (0.00 sec)

mysql> begin;
Query OK, 0 rows affected (0.01 sec)

mysql> insert into t (b) values (1), (2);
Query OK, 2 rows affected (0.00 sec)
Records: 2  Duplicates: 0  Warnings: 0

mysql> insert into t (b) values (3), (4);
Query OK, 2 rows affected (0.00 sec)
Records: 2  Duplicates: 0  Warnings: 0

mysql> select * from t;
+---------------------+------+
| a                   | b    |
+---------------------+------+
| 1441151880758558721 |    1 |
| 1441151880758558722 |    2 |
| 1441151880758558723 |    3 |
| 4899916394579099652 |    4 |
+---------------------+------+
4 rows in set (0.00 sec)
```

Comment (bb7133) [reply system-variables.md:2151]: I see, thanks!

**Review by bb7133 — APPROVED** [review]
LGTM

File: system-variables.md [hunk 4]
```diff
@@ -1554,6 +1554,16 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
 - Default value: `tikv,tiflash,tidb`
 - This variable is used to set the storage engine list that TiDB can use when reading data.
 
+### tidb_last_query_info <span class="version-mark">New in v4.0.14</span>
+
+- Scope: SESSION
+- Default value: ""
+- This is a read-only variable. It is used to query the transaction information of the last DML statement. The information includes:
+    - `txn_scope`: The scope of the transaction. It can be `global` or `local`.
```
Comment (TomShawn) [inline system-variables.md:1562]:
```suggestion
    - `txn_scope`: The scope of the transaction, which can be `global` or `local`.
```

File: system-variables.md [hunk 5]
```diff
@@ -1554,6 +1554,16 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
...
+    - `start_ts`: The start timestamp of the transaction.
+    - `for_update_ts`: The for_update_ts of the previously executed DML statement, which is an internal term of TiDB used for tests, usually, the end-user doesn't need to care about it.
```
Comment (TomShawn) [inline system-variables.md:1564]:
```suggestion
    - `for_update_ts`: The `for_update_ts` of the previously executed DML statement. This is an internal term of TiDB used for tests. Usually, you can ignore this information.
```

File: system-variables.md [hunk 6]
```diff
@@ -1554,6 +1554,16 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
...
+    - `error`: The error message if any.
```
Comment (TomShawn) [inline system-variables.md:1565]:
```suggestion
    - `error`: The error message, if any.
```

File: system-variables.md [hunk 7]
```diff
@@ -1554,6 +1554,16 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
 - Default value: `tikv,tiflash,tidb`
 - This variable is used to set the storage engine list that TiDB can use when reading data.
 
+### tidb_last_query_info <span class="version-mark">New in v4.0.14</span>
+
+- Scope: SESSION
+- Default value: ""
```
Comment (TomShawn) [inline system-variables.md:1560]: Does it persist to cluster?

File: system-variables.md [hunk 8]
```diff
@@ -2132,6 +2142,14 @@ explain select * from t where age=5;
...
 - Note that if `SHARD_ROW_ID_BITS` and `PRE_SPLIT_REGIONS` have been set when a table is created, the specified number of Regions are evenly split after the table creation.
 
+### `tidb_shard_allocate_step` <span class="version-mark">New in v5.0</span>
+
+- Scope: SESSION | GLOBAL
+- Persists to cluster: Yes
+- Default value: `9223372036854775807`
+- Range: `[1, 9223372036854775807]`
+- This variable controls the max number of the continous IDs. Generally, the allocating [`AUTO_RANDOM`](/auto-random.md) IDs or the [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md) annotated row IDs are incremental and continuous in one transaction. This variable can be used to solve the hotspot issue in the large transaction scenario.
```
Comment (TomShawn) [inline system-variables.md:2151]:
```suggestion
- This variable controls the maximum number of continuous IDs to be allocated for the [`AUTO_RANDOM`](/auto-random.md) or [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md) attribute. Generally, `AUTO_RANDOM` IDs or the `SHARD_ROW_ID_BITS` annotated row IDs are incremental and continuous in one transaction. You can use this variable to solve the hotspot issue in large transaction scenarios.
```

File: system-variables.md (hunk 7 above)
Comment (tangenta) [reply system-variables.md:1560]: `SESSION` scope system variables never persist to cluster.

File: system-variables.md [hunk 9]
```diff
@@ -1554,6 +1554,16 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
 - Default value: `tikv,tiflash,tidb`
 - This variable is used to set the storage engine list that TiDB can use when reading data.
 
+### tidb_last_query_info <span class="version-mark">New in v4.0.14</span>
+
+- Scope: SESSION
```
Comment (TomShawn) [inline system-variables.md:1559]:
```suggestion
- Scope: SESSION
- Persists to cluster: No
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**